homeassistant/components/systemmonitor/__init__.py

```diff
@@ -84,6 +88,14 @@ async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             entry, options=new_options, version=1, minor_version=2
         )
 
+        if entry.minor_version == 2:
+            new_options = {**entry.options}
+            if SENSOR_DOMAIN in new_options:
+                new_options.pop(SENSOR_DOMAIN)
```

Comment:
Generally we should not remove legacy items from the config entry data. There's no problem with leaving it like it is. Just stop using it.